### PR TITLE
fix: ShrinkWrapClassLoader finds ServiceProviders in WAR

### PR DESCRIPTION
**Pull request by DavideD**
Can this work?

The ShrinkwrapClassloader look for a specified path and if it cannot find the ndoe than look for the asset in WEB-INF/classes/

### CODE REVIEW

**Comment by DavideD** [comment]
I'm surprised to see my PR re-opened. I've rebased it to the latest changes in main if it's something that can be helpful.
It's been 10 years, so I don't remember anything about it :)

File: api/src/main/java/org/jboss/shrinkwrap/api/classloader/ShrinkWrapClassLoader.java [hunk 1]
```diff
@@ -175,6 +167,20 @@ public InputStream getInputStream() throws IOException {
 
                         }
 
+                        private Node node(final Archive<?> archive, final ArchivePath path) throws FileNotFoundException {
+                           Node node = archive.get(path);
+                           if (node != null)
+                              return node;
```
Comment (bartoszmajsak) [inline api/src/main/java/org/jboss/shrinkwrap/api/classloader/ShrinkWrapClassLoader.java:173]:
```suggestion
                           if (node != null) {
                              return node;
                           }
```

File: api/src/main/java/org/jboss/shrinkwrap/api/classloader/ShrinkWrapClassLoader.java [hunk 2]
```diff
@@ -175,6 +167,20 @@ public InputStream getInputStream() throws IOException {
...
+
+                           node = archive.get(ArchivePaths.create("WEB-INF/classes", path));
+                           if (node != null)
+                              return node;
```
Comment (bartoszmajsak) [inline api/src/main/java/org/jboss/shrinkwrap/api/classloader/ShrinkWrapClassLoader.java:177]:
```suggestion
                           if (node != null) {
                              return node;
                           }
```

**Review by bartoszmajsak — CHANGES_REQUESTED** [review]
Two minor suggestions. Thanks for taking care of the rebase!

**Review by bartoszmajsak — APPROVED** [review]
Thank you!

**Pull request merged**